### ISSUE TITLE
Refactor OrderedDict to dict

### ISSUE THREAD

**Issue by b-butler** [glotzerlab/signac-flow#323]
Python's `OrderedDict` class is known to be slower than the plain `dict`. Since 3.6 in CPython, `dict` objects have been ordered as well. This was made an API guarantee in 3.7 of the Python standard. However, within `flow` I do not believe we give an API guarantee of `dict` orderings anywhere. We do need the ordering as tests are currently written, but since tests aren't an API promise with the recent drop of 3.5 support, we can use the fact that in 3.6 we have the implementation order `dict` objects without needing the 3.7 standard for testing purposes.

Other opinions on this are welcome.

**Comment by bdice** [comment]
Thanks @b-butler! I misunderstood an earlier conversation and thought the order might be an API guarantee. Thanks for clarifying. This might be a good first issue!

**Comment by csadorf** [comment]
I thought that `OrderedDict` is literally a `dict` as of cpython 3.6. However, since the sort-order is only guaranteed as of Python 3.7, I'm not sure that this is true for PyPy as well.

**Comment by mikemhenry** [comment]
In 3.6 it was an implementation detail of CPython and formalized in the spec of 3.7 (as @csadorf  said). I did a bit of digging and have yet to find something definitive,  but it looks like CPython used what the PyPy team came up with for `dict` so it is also insertion ordered, see https://docs.python.org/3.6/whatsnew/3.6.html#new-dict-implementation and https://morepypy.blogspot.com/2015/01/faster-more-memory-efficient-and-more.html

tl;dr this is a good first issues and I bet since we dropped 3.5, changing `OrderedDict` to `dict` will still have all of our tests passing.

**Comment by b-butler** [comment]
On CPython 3.6.9, it seems that `dict` is a bit faster than `OrderedDict` (using some quick benchmarks on my local machine).

**Comment by bdice** [comment]
Done in #374.
